app/shop.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from datetime import datetime
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list
+    products: dict
+
+    def products_cost(self, cart: dict) -> dict:
+        return {
```

Comment:
The `products_cost` method returns a dictionary of product costs. Ensure that this aligns with how you intend to use these costs in other parts of the application, as seen in `customer.py`.